machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,389 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MCOConfig describes configuration for MachineConfigOperator.
+type MCOConfig struct {
```

Comment:
How fixed are these names?  With machine-config-ness already implied by the `machineconfiguration/v1` package name and `machineconfiguration.openshift.io` group name, should this be `OperatorConfig` or some such instead of `MCOConfig`?

If we cannot rely on the implicit namespacing, can we use `MachineConfigOperatorConfig` to make it more recognizable for folks unfamiliar with the acronym?  Saving typing doesn't seem like a big deal, and they each take about as long to read ;).

Reply:
This _will_ be OperatorConfig (I've commented elsewhere that we'll do a follow up). The reason MCO types are _complicated_ today is that we cannot move them all cause of external deps (ignition namely)